x/emissions/keeper/msgserver/msg_server_reputer_payload.go

```diff
@@ -103,3 +111,96 @@ func (ms msgServer) InsertReputerPayload(ctx context.Context, msg *types.InsertR
 
 	return &types.InsertReputerPayloadResponse{}, err
 }
+
+func (ms msgServer) validateValueBundle(ctx context.Context, valueBundle *types.InputValueBundle, topicId uint64, blockHeight int64) error {
+	networkInferences, err := ms.k.GetNetworkInferences(ctx, topicId, blockHeight)
+	if err != nil {
+		return errorsmod.Wrapf(err, "error getting inferences")
+	}
+
+	if err := validateWorkerValues(
+		networkInferences.InfererValues,
+		valueBundle.InfererValues,
+	); err != nil {
+		return fmt.Errorf("inferer values mismatch: %w", err)
```

Comment:
```suggestion
		return errorsmod.Wrapf(sdkerrors.ErrInvalidRequest, "inferer values mismatch: %w", err)
```

As the error returned here is the one that'll appear in the tx result it should be registered with a code. I propose to use the generic `sdkerrors.ErrInvalidRequest` for that

Reply:
makes sense, I'll do it above at the call site